multinet/db/models/table.py

```diff
@@ -72,6 +81,32 @@ def headers(self) -> List[str]:
 
         return keys
 
+    def get_metadata(self) -> Optional[ArangoEntityDocument]:
+        """Retrieve metadata for this table, if it exists."""
+        try:
+            doc = next(self.metadata_collection.find({"item_id": self.name}, limit=1))
+        except StopIteration:
+            return None
+
+        return ArangoEntityDocument(**doc)
+
+    def set_metadata(self, raw_data: Dict) -> ArangoEntityDocument:
+        """Set metadata for this table."""
+        try:
+            data = TableMetadata(**raw_data)
+        except PydanticValidationError:
+            raise InvalidMetadata(raw_data)
+
+        existing = self.get_metadata()
+        if existing is not None:
+            existing.table = data
+            doc_meta = self.metadata_collection.update(existing.dict())
+        else:
+            entity = EntityMetadata(item_id=self.name, table=data, graph=None)
+            doc_meta = self.metadata_collection.insert(entity.dict())
```

Comment:
Does this enable the PUT? What if we POST new data? We might want to overwrite the whole metadata object, not just update it.

Reply:
So just to clarify, `PUT` is intended to be used when creating/updating an entity, where as `POST` is intended to be used when creating new entities.

This code enables `PUT` at the table level, as in any existing table data is overwritten by the data passed into this function. You're correct that this function doesn't allow for updating the entire metadata object at once, which is by design. This class method is designed to only affect the table section of the metadata, and we don't really have a use case otherwise at the moment. I think in the future we'll update the API to better deal with metadata, but as a first approach, we're just focusing on table metadata.

Does that answer your question?

Reply:
What do you mean by "enable the PUT"?

Reply:
I'm not sure what @JackWilb meant but what I mean is replacing the existing data with the specified new data (so `PUT` as in set this entity to this value, regardless of what might already be there).

Reply:
@waxlamp : Sorry "enable PUT requests".

@AlmightyYakob: You've answered most of my question (the stuff related to PUT). However, I see that you use `.update()` here, which only updates the keys that are in `entity.dict`. Is it possible to remove a key using `update()`? I was wondering if we'd have a use case where we'd want to clear out keys

Reply:
I'm not sure if you can remove top level keys with `update`, but again this is only operating on the metadata for a table, scoped to the `table` key. Since this is done by creating a `TableMetadata` instance, any missing keys will take a value of whatever is default for that key (so likely `None` or some other default). So as far as Arango is concerned, every key will be updated, because the class instance has something for each of those keys, whether it's specified or default.

E.g. if you were to pass `{}` into the set metadata call, the result would be:

```
{
  ...,
  "table": {
    "columns": []
  }
}
```

Since the default for `columns` is `[]`.

Reply:
Perfect. I'll mark as resolved